src/lib.rs

```diff
@@ -558,44 +501,34 @@ fn create_texture_r8_static<R: Resources, F: Factory<R>>(
     width: u16,
     height: u16,
     data: &[u8],
-) -> Result<Texture<R>, TextureError> {
-    let texture = try!(factory.create_texture(tex::TextureInfo {
-        width: width,
-        height: height,
-        depth: 1,
-        levels: 1,
-        kind: tex::Kind::D2,
-        format: tex::R8,
-    }));
-    try!(factory.update_texture_raw(
-        &texture,
-        &(*texture.get_info()).into(),
-        data,
-        None,
-    ));
-    Ok(texture)
+) -> Result<gfx::handle::ShaderResourceView<R, f32>, CombinedError> {
+    let kind = tex::Kind::D2(width, height, tex::AaMode::Single);
+    let (_, texture_view) = try!(
+        factory.create_texture_const::<(gfx::format::R8, gfx::format::Unorm)>(kind, gfx::cast_slice(&data), false)
```

Comment:
I don't think you need a `cast_slice` here, since the format is `(R8, Unorm)` anyway